app/src/main/java/org/p2p/wallet/history/ui/historylist/HistoryListViewPresenter.kt

```diff
@@ -90,17 +130,15 @@ class HistoryListViewPresenter(
         }
     }
 
-    private fun attachToHistoryFlow() {
-        launch {
-            historyItemMapper.getHistoryAdapterItemFlow()
-                .filterNotNull()
-                .collect { items ->
-                    view?.showHistory(items)
-                    view?.showPagingState(PagingState.Idle)
-                }
+    private suspend fun getUserSendLinksCount(historyType: HistoryListViewType): Int {
+        return if (sendViaLinksToggle.isFeatureEnabled && historyType is HistoryListViewType.AllHistory) {
+            userSendLinksRepository.getUserLinksCount()
+        } else {
+            NO_LINKS_AVAILABLE_VALUE
         }
     }
 
+    @Throws(IllegalStateException::class)
     private fun handlePagingResult(result: HistoryPagingResult): List<HistoryTransaction> {
         return when (result) {
             is HistoryPagingResult.Error -> error(result.cause)
```

Comment:
that's weird. 
There is no reason to create the  `HistoryPagingResult` if we throw exception anyway.
You better remove `HistoryPagingResult` and throw exception from start

Reply:
`is HistoryPagingResult.Error -> error(result.cause) -> is HistoryPagingResult.Error -> throw result`

HistoryPagingResult comes from data layer, too much do rewrite